### PR TITLE
Detect and fix "Failed prop type" and "misspelled propType" React warnings

### PR DESCRIPTION
**Pull request by martinpitt**
See individual commits for details.

 - [x] Clean up React class creation (PR #10291)
 - [x] Fix missing `onChange` in pkg/docker/storage.jsx `DriveBox`
 - [x] Fix missing `emptyCaption` on Hardware Info page
 - [x] Fix wrong `emptyCaption` type in SETroubleshootPage Listing
 - [x] Fix wrong `hypervisorMax` on VCPUModalBody → InfoRecord
 - [x] Fix missing `onChange` on Firewall page

### CODE REVIEW

**Comment by martinpitt** [comment]
There is an [unrelated test failure](https://fedorapeople.org/groups/cockpit/logs/pull-10301-20181015-103030-6bb9a865-verify-fedora-29/log.html#242) here, I take the liberty to fix it en passant.

**Comment by martinpitt** [comment]
The [test log](https://fedorapeople.org/groups/cockpit/logs/pull-10301-20181015-103030-6bb9a865-verify-fedora-29/log) shows `Node environment: production`, so this didn't actually work. At least the docker test should have failed (as it does locally).

**Comment by martinpitt** [comment]
[This log](https://fedorapeople.org/groups/cockpit/logs/pull-10301-20181015-152424-f53644b7-verify-fedora-29/log.html) shows that this is now working. The `should have a unique "key" prop` messages appear, but are not fatal. But the failed prop types now are:
```
> error: Warning: Failed prop type: The prop `emptyCaption` is marked as required in `Listing`, but its value is `undefined`.
    in Listing (created by HardwareInfo)
    in HardwareInfo
[...]
RuntimeError: Warning: Failed prop type: The prop `emptyCaption` is marked as required in `Listing`, but its value is `undefined`.
```

**Comment by martinpitt** [comment]
I added two more commits to fix misspelled `propTypes`, these were easy to fix.

File: pkg/lib/cockpit-components-dialog.jsx [hunk 1]
```diff
@@ -236,10 +236,11 @@ class DialogFooter extends React.Component {
         );
     }
 }
-DialogFooter.PropTypes = {
+
+DialogFooter.propTypes = {
     cancel_clicked: PropTypes.func,
     cancel_caption: PropTypes.string,
-    actions: PropTypes.array,
+    actions: PropTypes.array.isRequired,
```
Comment (mvollmer) [inline pkg/lib/cockpit-components-dialog.jsx]: There is always the cancel button, no?  So "actions" could be optional and default to `[ ]`. But I think it's also okay to requite the caller to specify `[ ]` explicitly.

Comment (martinpitt) [reply pkg/lib/cockpit-components-dialog.jsx]: I wasn't aware of the implicit cancel button. Apparently right now, every caller specifies actions, but indeed we could dial this back when we have a case where only a Cancel button is appropriate. For error messages we generally use inline notifications though, so we don't do "error message" popups.

**Pull request merged**